### DISCUSSION TITLE
How to print TraceId in spring-cloud-gateway-4.x?

### DISCUSSION THREAD

**Discussion by Bigezno** [apache/skywalking#12272] [Q&A]
below is my project lib version：
SpringBoot：3.0.2，Spring Cloud Gateway：4.0.0，Skywalking：10.0.0
apm-toolkit-trace:9.2.0, apm-toolkit-logback-1.x：9.2.0，apm-toolkit-webflux-activation: 9.2.0

In gateway service，i can print the tid in my code when i use String traceId = TraceContext.traceId(); ,but logback cannot print tid,only show [TID:N/A]. 
![image](https://github.com/apache/skywalking/assets/24370504/25ec186c-9f27-466c-8375-339afd7a3d1c)

The java-agent in website: 
![image](https://github.com/apache/skywalking/assets/24370504/d16cc5c3-315a-4ee3-8d78-8dade0287ef1)

**Comment by Bigezno** [comment]
@wu-sheng ,  Could you help me?

**Reply by wu-sheng** [reply]
I don't know why you ping me, this feature has nothing related to me. 

I don't use Spring gateway at all. Maybe this doc? https://skywalking.apache.org/docs/skywalking-java/latest/en/setup/service-agent/java-agent/application-toolkit-webflux/

**Comment by CzyerChen** [comment]
`but logback cannot print tid,only show [TID:N/A].`

Can other logback calls print traceId correctly?

**Reply by CzyerChen** [reply]
You can follow [this issue](https://github.com/apache/skywalking/issues/11205) to check your configuration, and maybe show more details about your code and config.

**Comment by Bigezno** [comment]
The Skywalking -Java-agent lib version:
activations folder: 
 
![image](https://github.com/apache/skywalking/assets/24370504/bc610ed6-ccd4-48dc-8779-d9cf6276a758)

plugins folder:
![image](https://github.com/apache/skywalking/assets/24370504/b97bab05-7bc3-42c7-9135-944ae13d6ac8)

Logback config:
![image](https://github.com/apache/skywalking/assets/24370504/84e599cb-fb82-42cb-9a17-c8c800aec43b)

TraceId Log pom:
![image](https://github.com/apache/skywalking/assets/24370504/b037141e-1f29-4cb7-9998-22f1fe4c070c)

when I invoke a api interface, The log is printed as follows

![image](https://github.com/apache/skywalking/assets/24370504/f52c3c93-ecb6-436d-831b-ce828b8563f6)

@CzyerChen

**Reply by Bigezno** [reply]
The Maven Pom:
![image](https://github.com/apache/skywalking/assets/24370504/8e13f7ea-a73a-4aaa-81fc-f3dc1514aa7e)

**Comment by Bigezno** [comment]
And when I set Mdc in tid:
![image](https://github.com/apache/skywalking/assets/24370504/2e287e3c-fe17-4d76-8c80-b2178441201e)
![Uploading image.png…]()

**Reply by Bigezno** [reply]
![image](https://github.com/apache/skywalking/assets/24370504/abf2deca-8c70-47d5-a1fe-b595d6006acd)
![image](https://github.com/apache/skywalking/assets/24370504/5095c544-8dec-42fe-8df8-0f252b9e23e5)

**Comment by Bigezno** [comment]
Is there another way to print traceId ? un... do not set tid to mdc

**Reply by HongboSuo** [reply]
I also have this problem, have you solved it?

**Reply by LightShine** [reply]
> I also have this problem, have you solved it?

how to solve this problem?
